File: Test/TestModel.m
//
//  TestModel.m
//  MuscleAssert
//
//  Created by akuraru on 2017/01/05.
//
//

#import "TestModel.h"

@implementation TestModel

- (instancetype)initWithString:(NSString *)string {
    return [self initWithString:string number:@0 interger:0];
}

- (instancetype)initWithString:(NSString *)string number:(NSNumber *)number interger:(NSInteger)interger {
    self = [super init];
    if (self) {
        _string = string;
        _number = number;
        _integer = interger;
    }
    return self;
}

- (BOOL)isEqual:(id)object {
    return [self.string isEqualToString:[object string]]
    && [self.number isEqualToNumber:[object number]]
    && self.integer == self.integer;
}

- (NSString *)debugDescription {
    return [NSString stringWithFormat:@"TestModel(string: %@, number: %@, integer: %zd)", self.string, self.number, self.integer];
}

@end

File: SwiftTest/TestModel/TestModel.swift
//
//  TestModel.swift
//  SwiftTest
//
//  Created by akuraru on 2017/10/21.
//

import Foundation

class TestModel: NSObject {
    let string: String
    let integer: Int
    
    init(string: String = "", integer: Int = 0) {
        self.string = string
        self.integer = integer
    }
    
    override func isEqual(_ object: Any?) -> Bool {
        guard let object = object as? TestModel else {
            return false;
        }
        return self.string == object.string &&
            self.integer == object.integer
    }
    
    override var description: String {
        return "TestModel(string: \(string), integer: \(integer))"
    }
}
